components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1181,12 +1183,690 @@ private class MonitoredItemWrapper {
                 return fieldName;
             }
 
+            internal EventFilter BuildBasicEventFilter(Session session) {
+                var simpleEventTypeNodeId = NodeId.Parse(EventTemplate.EventFilter.EventTypeDefinitionId);
+                var fields = ClientUtils.CollectInstanceDeclarationsForType(session, simpleEventTypeNodeId);
+
+                var eventFilter = new EventFilter();
+                foreach (var field in fields) {
+                    eventFilter.SelectClauses.Add(new SimpleAttributeOperand() {
```

Comment:
There is a AddSelectClause method on the EventFilter class. Should we use that?

Reply:
Good catch. I'll fix.

Reply:
AddSelectClause doesn't support BrowsePath as a collection. Need to be explicit.